src/sagas/activity/sagas.js

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable */
+
+import uuid from 'uuid4';
+import { call, put, take } from 'redux-saga/effects';
+
+import { doneFetchMessages, messageAdded } from '../../actions/activity';
+import defaultDB from '../../services/database';
+import { convertFromDatabase, convertToDatabase } from '../../utils/mapping/activity';
+import type { ActivityLogMessage } from '../../types/ActivityLogMessage';
+import { createDatabaseUpdateChannel } from '../database';
+
+export function* fetchMessagesSaga(action) {
+  try {
+    const db = yield defaultDB;
+    const messages = db.objects('MessageJob');
+    yield put(doneFetchMessages(messages));
+  } catch (error) {
+    console.log(`Activity messages fetch failed with error: ${error.toString()}`);
+  }
+}
+
+export function* watchNewMessages(): Generator<*, *, *> {
```

Comment:
Since we're providing multiple accounts support, we should also filter our models, so that select only ones related to current account. You can use `currentAccountBasedUpdate` from `src/sagas/accounts/sagas.js` for that purpose. It will handle subscribing and will call callback on each related change. 

Also, you need to add account field into message database model and likely add inverse relation on account model to simplify fetch. Please update `addNewNessageSaga` correspondingly.

Reply:
@seland I am unclear with this logic. Can you help me to get this sorted?